staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -405,6 +411,17 @@ func (c completedConfig) New(delegationTarget DelegationTarget) (*GenericAPIServ
 		s.postStartHooks[k] = v
 	}
 
+	genericApiServerHookName := "generic-apiserver-start-informers"
+	if !s.wasHookRegistered(genericApiServerHookName) {
```

Comment:
Add the nil check here so we don't register the post start hook at all if its missing.